src/mappings/erc20Templates.ts

```diff
@@ -235,40 +292,28 @@ export function handleProviderFee(event: ProviderFee): void {
     event.params.validUntil
   }"}`
 
-  const orderId = getOrderId(
+  const order = searchOrderForEvent(
     event.transaction.hash.toHex(),
     event.address.toHex(),
-    event.transaction.from.toHex()
+    event.transaction.from.toHex(),
+    event.logIndex.toI32()
   )
-  const order = Order.load(orderId)
 
   if (order) {
     order.providerFee = providerFee
     order.providerFeeValidUntil = event.params.validUntil
     order.save()
     return
   }
-
-  let orderReuse = OrderReuse.load(event.transaction.hash.toHex())
+  const orderReuse = searchOrderReusedForEvent(
+    event.transaction.hash.toHex(),
+    event.address.toHex(),
+    event.logIndex.toI32()
+  )
   if (orderReuse) {
+    log.info('order reuse id in provider fee handler: {}', [orderReuse.id])
     orderReuse.providerFee = providerFee
     orderReuse.providerFeeValidUntil = event.params.validUntil
     orderReuse.save()
-  } else {
-    orderReuse = new OrderReuse(event.transaction.hash.toHex())
```

Comment:
Why did you delete this?

Reply:
In this case either `OrderStarted`, either `OrderReuse` must be fetched before `ProviderFee` event. They must already exist in subgraph.

cc @mihaisc